pkgs/development/libraries/SDL_compat/default.nix

```diff
@@ -25,25 +28,39 @@ stdenv.mkDerivation rec {
     hash = "sha256-PDGlMI8q74JaqMQ5oX9Zt5CEr7frFQWECbuwq5g25eg=";
   };
 
+  patches = optionals stdenv.hostPlatform.isDarwin [
+    # Fix broken install name in dylib, https://github.com/libsdl-org/sdl12-compat/issues/194
+    # Remove when bump > 1.2.52
+    (fetchpatch {
+      name = "SDL_compat-fix-install-name.patch";
+      url = "https://github.com/libsdl-org/sdl12-compat/commit/5642d47ae489f2eb362cece2464ecc466a700ed5.patch";
+      sha256 = "sha256-kQ5H8gXjTZVHHRF6UpgXKl3NIy87iULcc2iCsYt5Hqo=";
+    })
+  ];
+
   nativeBuildInputs = [ cmake pkg-config ];
 
   propagatedBuildInputs = [ SDL2 ]
+    ++ optionals stdenv.hostPlatform.isDarwin [ libiconv Cocoa ]
     ++ optionals openglSupport [ libGL libGLU ];
 
   enableParallelBuilding = true;
 
   setupHook = ../SDL/setup-hook.sh;
 
   postFixup = ''
-    for lib in $out/lib/*.so* ; do
+    for lib in $out/lib/*${stdenv.hostPlatform.extensions.sharedLibrary}* ; do
```

Comment:
Shouldn't this be `targetPlatform` instead/

Reply:
I haven't specifically tested cross-compiling, where the nuances of `*Platform` would become relevant, but `targetPlatform` should only be relevant when compiling something that produces something for another platform (like a compiler).

https://nixos.org/manual/nixpkgs/stable/#ssec-cross-platform-parameters

`buildPlatform` is the platform we're building X on, `hostPlatform` the platform we're building X for and `targetPlatform` the platform X itself builds for. To my understanding, `host` should be correct one here.

Reply:
I stand corrected!